where-to-travel/src/main/webapp/script.js

```diff
@@ -414,79 +453,125 @@ function populatePlaces(placeArray) {
+        toggleFocusOff();
+        focusedPin = placeMarker;
+        selectLocationCard(placeMarker.getTitle());
+        placeMarker.setIcon(SELECTED_PIN_PATH);
+        focusedCard.scrollIntoView({behavior: 'smooth', block: 'center'});
+      });
 
-    placeMarker.addListener('mouseover', function () {
-      placeMarker.setIcon(SELECTED_PIN_PATH);
-    });
+      placeMarker.addListener('mouseover', function () {
+        placeMarker.setIcon(SELECTED_PIN_PATH);
+      });
 
-    placeMarker.addListener('mouseout', function () {
-      if (placeMarker != focusedPin) {
-        placeMarker.setIcon(PIN_PATH);
-      }
-    });
+      placeMarker.addListener('mouseout', function () {
+        if (placeMarker != focusedPin) {
+          placeMarker.setIcon(PIN_PATH);
+        }
+      });
 
-    markers.push(placeMarker);
+      markers.push(placeMarker);
+    }
   }
 
   document.getElementById(SCROLL_ID).hidden = false;
-
   // Handle favoriting a place
   $('.icon').click(function() {
+    const name = $(this).parent().parent().parent().attr('placeName');
+    const placeId = $(this).parent().next().next().next().attr('savedPlaceId');
+    let card = document.getElementById(name);
     $(this).toggleClass('press');
     if (firebase.auth().currentUser && $(this).hasClass('press')) {
-      const name = $(this).parent().parent().parent().attr('placeName');
-      const link = $(this).parent().next().attr('href');
       const time = $(this).parent().next().next().text();
        // Add users saved places to the real time database in Firebase when star is pressed
       const database = firebase.database();
       var uID = firebase.auth().currentUser.uid;
-      var ref = database.ref('users/' + uID + '/' + 'places' + '/' + name);
+      var ref = database.ref('users/' + uID + '/places/' + name);
       var data = {
         name: name,
-        link: link,
-        time: time,
+        timeAsString: time,
+        timeInSeconds: card.getAttribute('data-timeInSeconds'),
+        place_id: placeId,
+      }
+      ref.set(data);
+      // Store the lat/lng of each place in the database under geometry/location 
+      // to be the same as the place object created form the search.
+      let lat = card.dataset.lat;
+      let lng = card.dataset.lng;
+      var ref = ref + '/geometry/location/';
+      var data = {
+        lat: lat,
+        lng: lng
       }
       ref.set(data);
+      savedPlacesSet.add(placeId);
     } else if (firebase.auth().currentUser && (!$(this).hasClass('press'))) {
-      const name = $(this).parent().parent().parent().attr('placeName');
-       // Delete user saved places when the star is not pressed
-      var ref = firebase.database().ref('users/' + firebase.auth().currentUser.uid + '/' + 'places' + '/' + name);
+      // Delete user saved places when the star is not pressed/unpressed
+      var ref = firebase.database().ref('users/' + firebase.auth().currentUser.uid + '/places/' + name);
```

Comment:
This url seems to be used frequently. Would it be worth turning into a function?